src/deepymod/model/deepmod.py

```diff
@@ -38,143 +37,150 @@ def forward(self, input: Tuple[TensorList, TensorList]) -> Tuple[TensorList, Ten
 
         Returns:
-            coefficients (np.ndarray)
+            (np.ndarray): Coefficients of size (n_samples, n_outputs)
         """
         pass
 
 
 class Library(nn.Module):
-    """ Abstract class that calculates the library function and time derivatives. 
-
-    Args:
-        nn (PyTorch Class): Module of the function approximator, typically a neural network. 
-    """
     def __init__(self) -> None:
-        super().__init__()  
+        """Abstract baseclass for the library module."""
+        super().__init__()
         self.norms = None
 
-    def forward(self, input: Tuple[TensorList, TensorList]) -> Tuple[TensorList, TensorList]:
-        """Compute the library (time derivatives and thetas) given a dataset
+    def forward(self, input: Tuple[torch.Tensor, torch.Tensor]) -> Tuple[TensorList, TensorList]:
+        """Compute the library (time derivatives and thetas) from a given dataset. Also calculates the norms
+        of these, later used to calculate the normalized coefficients.
 
         Args:
-            input (Tuple[TensorList, TensorList]): tuple with firstly the predicted values and 
-            secondly the respective coordinates for the prediction
+            input (Tuple[TensorList, TensorList]): (prediction, data) tuple of size ((n_samples, n_outputs), (n_samples, n_dims))
 
         Returns:
-            Tuple[TensorList, TensorList]: The time derivatives and thetas
+            Tuple[TensorList, TensorList]: Temporal derivative and libraries of size ([(n_samples, 1) x n_outputs]), [(n_samples, n_features)x n_outputs])
         """
+
         time_derivs, thetas = self.library(input)
         self.norms = [(torch.norm(time_deriv) / torch.norm(theta, dim=0, keepdim=True)).detach().squeeze() for time_deriv, theta in zip(time_derivs, thetas)]
         return time_derivs, thetas
 
     @abstractmethod
-    def library(self, input: Tuple[torch.Tensor, torch.Tensor]) -> Tuple[TensorList, TensorList]: 
-        """Abstract method that should compute the library:
-        
+    def library(self, input: Tuple[torch.Tensor, torch.Tensor]) -> Tuple[TensorList, TensorList]:
+        """Abstract method. Specific method should calculate the temporal derivative and feature matrices.
+        These should be a list; one temporal derivative and feature matrix per output.
+
         Args:
-            input (Tuple[TensorList, TensorList]): predictions and coordinates for which to evaluate 
-            the library.
-            
+        input (Tuple[TensorList, TensorList]): (prediction, data) tuple of size ((n_samples, n_outputs), (n_samples, n_dims))
+
         Returns:
-            (Tuple[TensorList, TensorList]): The time derivatives and thetas
+        Tuple[TensorList, TensorList]: Temporal derivative and libraries of size ([(n_samples, 1) x n_outputs]), [(n_samples, n_features)x n_outputs])
         """
         pass
 
 
 class DeepMoD(nn.Module):
```

Comment:
Can this class be at the very top of the code? To give some 'hierarchy' of importance - I found it a bit confusing when read it for the first time

Reply:
I agree; but we're moving the classes above to a separate file soon, so let's leave this for now.